### PR TITLE
Config revamp

### PR DESCRIPTION
**Pull request by erezmus**

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -0,0 +1,90 @@
...
+
+
+#
+# Stage: Production environment
+#
+FROM nginx:stable-alpine@sha256:0dfc8450deb8c7f06fbaac27e453ac3262df7d3a93639c4e2f48ee39434ec017 as prod
+ENV NODE_ENV=production
+
+COPY --from=yarn-prod /app/ .
+COPY --from=build-prod /app/dist/ dist/
```
Comment (erkannt) [inline Dockerfile:85]: Can we maybe reduce the number of build stage redirect? e.g. base feels redundant to node.

File: Makefile [hunk 2]
```diff
@@ -1,55 +1,76 @@
...
-DOCKER_COMPOSE = IMAGE_TAG=${IMAGE_TAG} docker-compose -f docker-compose.build.yml
 
-yarn:
+DOCKER_COMPOSE = IMAGE_TAG=${IMAGE_TAG} docker-compose
+DOCKER_COMPOSE_TEST = IMAGE_TAG=${IMAGE_TAG} docker-compose -f docker-compose.yml -f docker-compose.test.yml
+DOCKER_COMPOSE_CI = IMAGE_TAG=${IMAGE_TAG} docker-compose -f docker-compose.yml -f docker-compose.ci.yml
+DOCKER_COMPOSE_BUILD = IMAGE_TAG=${IMAGE_TAG} docker-compose -f docker-compose.build.yml
+
+help:
+	@echo "Apptivio"
```
Comment (erkannt) [inline Makefile:12]: 'Apptivio' ?

File: Makefile [hunk 3]
```diff
@@ -1,55 +1,76 @@
...
+	${DOCKER_COMPOSE} logs -f reviewer-client
+
+start_ci: ## start with production build, with reviewer-mocks mocking api, continuum-adaptor and continuum
+	${DOCKER_COMPOSE_CI} pull reviewer-mocks
+	${DOCKER_COMPOSE_CI} up -d reviewer-mocks reviewer-client 
+	./.scripts/docker/wait-healthy.sh reviewer-client_app 20
+	./.scripts/docker/wait-healthy.sh reviewer-client_mocks 20
+
+stop: ## stop all services
+	${DOCKER_COMPOSE} down
```
Comment (erkannt) [inline Makefile:58]: Maybe also delete network as part of cleanup?

File: config/config.public.example.json [hunk 4]
```diff
@@ -1,4 +1,4 @@
 {
     "client_login_url": "http://localhost:3003/submit",
-    "client_api_url": "http://localhost:9000"
+    "client_api_url": "http://localhost:3003"
 }
```
Comment (erkannt) [inline config/config.public.example.json:4]: This file can be removed entirely, right?

File: deployment/nginx.conf.template [hunk 5]
```diff
@@ -19,15 +19,11 @@ http {
...
-      alias ${PUBLIC_CONFIG_PATH};
+      proxy_pass ${CLIENT_TOKEN_EXCHANGE_PROXY_URL};
     }
 
     location / {
-      root /usr/share/nginx/html;
+      root /app/dist;
       try_files $uri /index.html;
     }
   }
```
Comment (erkannt) [inline deployment/nginx.conf.template:29]: I think this file can also be removed, right?

File: docker-compose.yml [hunk 6]
```diff
@@ -1,15 +1,61 @@
...
+  nginx:
+    container_name: reviewer-client_nginx
+    image: nginx:stable-alpine@sha256:0dfc8450deb8c7f06fbaac27e453ac3262df7d3a93639c4e2f48ee39434ec017
+    networks:
+      - "reviewer"
+    ports:
+      - '9000:9000'
+    volumes:
+      - ./config/nginx/nginx.dev.conf:/etc/nginx/nginx.conf
+
```
Comment (erkannt) [inline docker-compose.yml:41]: Maybe best to be consistent between the compose files and always use a separate nginx container? Not sure myself.

For digest pinning I'd say its only strictly needed for docker-compose.ci.yaml. But maybe also best to have the same image for all.

File: scripts/build-config.sh [hunk 7]
```diff
@@ -1,23 +1,11 @@
...
-sed -i "s~\${CLIENT_AUTH_PROXY_URL}~$client_auth_proxy_url~g" /etc/nginx/nginx.conf
-sed -i "s~\${PUBLIC_CONFIG_PATH}~$PUBLIC_CONFIG_PATH~g" /etc/nginx/nginx.conf
+sed -i "s~\${CLIENT_PORT}~$CLIENT_PORT~g" /etc/nginx/nginx.conf
+sed -i "s~\${CLIENT_API_PROXY_URL}~$CLIENT_API_PROXY_URL~g" /etc/nginx/nginx.conf
+sed -i "s~\${CLIENT_TOKEN_EXCHANGE_PROXY_URL}~$CLIENT_TOKEN_EXCHANGE_PROXY_URL~g" /etc/nginx/nginx.conf
+
+env
+cat /etc/nginx/nginx.conf
 
 nginx -g 'daemon off;';
```
Comment (erkannt) [inline scripts/build-config.sh:11]: Another file that should no longer be needed if I'm not mistaken.

File: webpack.parts.js [hunk 8]
```diff
@@ -190,8 +158,8 @@ exports.generateSourceMaps = ({ type }) => ({
 exports.newRelic = () => ({
     plugins: [
         new HtmlInjectNewRelicPlugin({
-            license: infraConfig.new_relic_client_license_key,
-            applicationID: infraConfig.new_relic_client_app_id,
+            license: process.env.NEW_RELIC_CLIENT_LICENSE_KEY,
+            applicationID: process.env.NEW_RELIC_CLIENT_APP_ID
```
Comment (erkannt) [inline webpack.parts.js:162]: Are these baked into the client image? I am unclear where these should go/are needed.

File: Dockerfile [hunk 9]
```diff
@@ -0,0 +1,90 @@
+FROM node:12-alpine@sha256:5646d1e5bc470500414feb3540186c02845db0e0e1788621c271fbf3a0c1830d AS node
+ARG image_tag=latest
+ENV NODE_OPTIONS --unhandled-rejections=strict
+WORKDIR /app
+
+LABEL maintainer="eLife Reviewer Product Team <reviewer-product@elifesciences.org>"
```
[the remark this answers is not in the document]
Comment (erezmus) [reply Dockerfile]: done

File: Dockerfile [hunk 10]
```diff
@@ -0,0 +1,90 @@
...
+
+RUN yarn run build
+
+
+
+#
+# Stage: Production environment
+#
+FROM nginx:stable-alpine@sha256:0dfc8450deb8c7f06fbaac27e453ac3262df7d3a93639c4e2f48ee39434ec017 as prod
+ENV NODE_ENV=production
```
[the remark this answers is not in the document]
Comment (erezmus) [reply Dockerfile]: fixed

File: Dockerfile [hunk 11]
```diff
@@ -0,0 +1,90 @@
...
+
+
+#
+# Stage: Development NPM install
+#
+FROM yarn-prod AS yarn-dev
+
+RUN yarn install
+
+
```
[the remark this answers is not in the document]
Comment (erezmus) [reply Dockerfile]: fixed

File: config/config.public.example.json (hunk 4 above)
Comment (erezmus) [reply config/config.public.example.json]: removed

File: webpack.parts.js (hunk 8 above)
Comment (erezmus) [reply webpack.parts.js:162]: we'll clean this up later

File: scripts/build-config.sh (hunk 7 above)
Comment (erezmus) [reply scripts/build-config.sh]: removed

File: Dockerfile (hunk 1 above)
Comment (erezmus) [reply Dockerfile:68]: fixed

File: deployment/nginx.conf.template (hunk 5 above)
Comment (erezmus) [reply deployment/nginx.conf.template:34]: done

**Pull request merged**